scripts/pi-hole/js/queries.js

```diff
@@ -496,28 +432,108 @@ $(function () {
   });
 
   $("#resetButton").click(function () {
-    tableApi.search("").draw();
-    $("#resetButton").addClass("hidden");
+    tableApi.search("");
+    resetColumnsFilters();
   });
 
   // Disable autocorrect in the search box
   var input = document.querySelector("input[type=search]");
-  input.setAttribute("autocomplete", "off");
-  input.setAttribute("autocorrect", "off");
-  input.setAttribute("autocapitalize", "off");
-  input.setAttribute("spellcheck", false);
-
-  var chkboxData = localStorage.getItem("query_log_filter_chkbox");
-  if (chkboxData !== null) {
-    // Restore checkbox state
-    $("#autofilter").prop("checked", chkboxData === "true");
+  if (input !== null) {
+    input.setAttribute("autocomplete", "off");
+    input.setAttribute("autocorrect", "off");
+    input.setAttribute("autocapitalize", "off");
+    input.setAttribute("spellcheck", false);
+  }
+});
+
+function addColumnFilter(event, colID, filterstring) {
+  // Don't do anything when NOT explicitly requesting multi-selection functions
+  if (!event.ctrlKey && !event.metaKey && !event.shiftKey) {
+    return;
+  }
+
+  if (event.shiftKey) {
+    filterstring = "";
+  }
+
+  tableFilters[colID] = filterstring;
+
+  applyColumnFiltering();
+}
+
+function resetColumnsFilters() {
+  tableFilters.forEach(function (value, index) {
+    tableFilters[index] = "";
+  });
+
+  // Clear filter reset button
+  applyColumnFiltering();
+}
+
+function applyColumnFiltering() {
+  var showReset = false;
+  tableFilters.forEach(function (value, index) {
+    // Prepare regex filter string
+    var regex = "";
+
+    // Split filter string if we received a combined ID#Name column
+    var valArr = value.split("#");
+    if (valArr.length > 0) {
+      value = valArr[0];
+    }
+
+    if (value.length > 0) {
+      // Exact matching
+      regex = "^" + value + "$";
+
+      // Add background color
+      tableApi.$("td:eq(" + index + ")").addClass("highlight");
```

Comment:
The highlight color is too light...

Reply:
This is personal taste. It is the same color we use also in the domain list when highlighting a regex match. The color was discussed in the Discourse thread and I think we agreed that it is clear enough. You also don't want to make reading the cell content difficult.

![Screenshot from 2020-06-03 19-46-30](https://user-images.githubusercontent.com/16748619/83670184-1196cf00-a5d3-11ea-8573-b5ad9e5a9870.png)

Reply:
Agree. I think the color is rather obvious (for non color-blind at least)

Reply:
Just switched to the branch and saw that the filter color for me is indeed quite aggressiv. Is there a reason why I have this strong yellow and you the sand colored? I remember that I used to have the sand color also when I tried the branch last time (27.04.)

![Bildschirmfoto zu 2020-06-03 22-30-24](https://user-images.githubusercontent.com/26622301/83687239-c76e1780-a5eb-11ea-9bae-bf7d91addc9f.png)